src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -230,6 +230,12 @@ internal TabItem AddExtensionTabItem(IViewExtension viewExtension, ContentContro
                 else
                 {
                     tab.Content = contentControl.Content;
+                    var extensionWindow = contentControl as Window;
+                    if (extensionWindow != null)
+                    {
+                        // Make sure the extension window closes with Dynamo
+                        extensionWindow.Owner = this;
```

Comment:
Interesting.. I start to think this maybe why the fix did not work for me because when I tried it, I was calling close() specifically for Tuneup while your code should work for both TuneUp and NodePropertyPalette.

Reply:
However, giving this more thoughts. I think this approach is excellent that Dynamo should handle it properly. We may still want to release a version of Tuneup which contain a close() call so that current users who are using Tuneup can be saved.

Reply:
Yes, that's why I opened https://github.com/DynamoDS/TuneUp/pull/33. We should publish a new version after we merge that one.

Reply:
@mmisol Gotcha